svir/dialogs/viewer_dock.py

```diff
@@ -654,12 +658,15 @@ def layer_changed(self):
             elif self.output_type == 'recovery_curves':
                 fill_fields_multiselect(
                     self.fields_multiselect, self.iface.activeLayer())
+            else:  # no plots for this layer
+                self.current_selection = {}
             if self.iface.activeLayer().selectedFeatureCount() > 0:
                 self.set_selection()
 
     def remove_connects(self):
         try:
-            self.iface.activeLayer().selectionChanged.disconnect(self.redraw)
+            self.iface.activeLayer().selectionChanged.disconnect(
+                self.set_selection)
         except (TypeError, AttributeError):
             pass
```

Comment:
Could you add a comment saying when a TypeError is expected and when an AttributeError is expected?